protected/controllers/CsvController.php

```diff
@@ -0,0 +1,240 @@
+            array('var_name' => 'qualifications'  , 'default' => null,),
+            array('var_name' => 'contact_label_id', 'default' => null,),
+        );
+
+        foreach ($contact_cols as $col){
+            $contact->$col['var_name'] = isset($patient[$col['var_name']]) ? $patient[$col['var_name']] : $col['default'];
+        }
+
+        if(!$contact->save()){
+            return $contact->getErrors();
+        }
+
+        $address = new Address();
+        $address_cols = array(
+            array('var_name' => 'address1'       , 'default' => null,),
+            array('var_name' => 'address2'       , 'default' => null,),
+            array('var_name' => 'city'           , 'default' => null,),
+            array('var_name' => 'postcode'       , 'default' => null,),
+            array('var_name' => 'county'         , 'default' => null,),
+            array('var_name' => 'country_id'     , 'default' => 15,),
+            array('var_name' => 'email'          , 'default' => null,),
+            array('var_name' => 'date_start'     , 'default' => null,),
+            array('var_name' => 'date_end'       , 'default' => null,),
+            array('var_name' => 'address_type_id', 'default' => null,),
+        );
+
+        foreach ($address_cols as $col){
+            $address->$col['var_name'] = isset($patient[$col['var_name']]) ? $patient[$col['var_name']] : $col['default'];
+        }
+        $address->contact_id = $contact->id;
+
+        if(!$address->save()){
+            return $address->getErrors();
+        }
+
+        $new_patient = new Patient();
+        $patient_cols = array(
+            array('var_name' => 'pas_key'                       , 'default' => null,),
+            array('var_name' => 'dob'                           , 'default' => null,),
+            array('var_name' => 'gender'                        , 'default' => 'U',),
+            array('var_name' => 'nhs_num'                       , 'default' => null,),
+            array('var_name' => 'gp_id'                         , 'default' => null,),
+            array('var_name' => 'date_of_death'                 , 'default' => null,),
+            array('var_name' => 'practice_id'                   , 'default' => null,),
+            array('var_name' => 'ethnic_group_id'               , 'default' => null,),
+            array('var_name' => 'archive_no_allergies_date'     , 'default' => null,),
+            array('var_name' => 'archive_no_family_history_date', 'default' => null,),
+            array('var_name' => 'archive_no_risks_date'         , 'default' => null,),
+            array('var_name' => 'deleted'                       , 'default' => null,),
+            array('var_name' => 'nhs_num_status_id'             , 'default' => null,),
+            array('var_name' => 'is_deceased'                   , 'default' => null,),
+            array('var_name' => 'is_local'                      , 'default' => null,),
```

Comment:
Should this be set to 1 so they can modify the patient details? Or should we leave it null by default?